--- /src/MainWindow.vala	2018-11-17 18:41:11.000000000 +0300
+++ /src/MainWindow.vala	2019-02-04 23:36:26.349104088 +0300
@@ -624,10 +624,6 @@
         }
 
         private void header_build_style_switcher () {
-            if (PlayMyMusicApp.instance.get_os_info ("PRETTY_NAME").index_of ("elementary") == -1) {
-                return;
-            }
-
             var mode_switch = new Granite.ModeSwitch.from_icon_name ("display-brightness-symbolic", "weather-clear-night-symbolic");
             mode_switch.valign = Gtk.Align.CENTER;
             mode_switch.active = settings.use_dark_theme;
