--- /filechooser-portal/ExternalWindow.vala
+++ /filechooser-portal/ExternalWindow.vala
@@ -34,6 +34,7 @@ public interface ExternalWindow : GLib.O
                 warning ("Error getting external X11 window: %s", e.message);
                 return null;
             }
+#if WAYLAND
         } else if (handle.has_prefix (WAYLAND_PREFIX)) {
             try {
                 external_window = new ExternalWindowWayland (handle.substring (WAYLAND_PREFIX.length));
@@ -41,6 +42,7 @@ public interface ExternalWindow : GLib.O
                 warning ("Error getting external Wayland window: %s", e.message);
                 return null;
             }
+#endif
         } else {
             warning ("Unhandled parent window type %s", handle);
         }
@@ -92,6 +94,7 @@ public class ExternalWindowX11 : Externa
     }
 }
 
+#if WAYLAND
 public class ExternalWindowWayland : ExternalWindow, GLib.Object {
     private static Gdk.Display? wayland_display = null;
 
@@ -128,3 +131,4 @@ public class ExternalWindowWayland : Ext
         }
     }
 }
+#endif
--- /filechooser-portal/meson.build
+++ /filechooser-portal/meson.build
@@ -1,5 +1,14 @@
 libexec_dir = join_paths(get_option('prefix'), get_option ('libexecdir'))
 
+fc_deps = [
+    pantheon_files_widgets_dep,
+    dependency('gdk-x11-3.0'),
+    project_config_dep
+]
+if wayland_dep.found ()
+  fc_deps += wayland_dep
+endif
+
 executable(
     'io.elementary.files.xdg-desktop-portal',
     'ExternalWindow.vala',
@@ -7,12 +16,7 @@ executable(
     'LegacyFileChooserDialog.vala',
     'Main.vala',
 
-    dependencies : [
-        pantheon_files_widgets_dep,
-        dependency('gdk-x11-3.0'),
-        dependency('gdk-wayland-3.0'),
-        project_config_dep
-    ],
+    dependencies : fc_deps,
     install: true,
     install_dir: libexec_dir,
 )
--- /meson.build
+++ /meson.build
@@ -83,6 +83,12 @@ add_project_arguments(
     language: 'vala'
 )
 
+wayland_dep = dependency('gdk-wayland-3.0', required: false)
+
+if wayland_dep.found ()
+  add_project_arguments ('--define=WAYLAND', language: 'vala')
+endif
+
 
 #Configuration file
 
