--- /filechooser-portal/meson.build	2021-08-24 16:59:14.592537404 +0300
+++ /filechooser-portal/meson.build	2021-08-24 16:40:40.475509115 +0300
@@ -55,16 +55,16 @@
     install_dir: join_paths(get_option('prefix'), get_option('datadir'), 'dbus-1', 'services')
 )
 
-custom_target(
-    'portal-tester',
-    command: [
-        'flatpak-builder',
-        '--user',
-        '--install',
-        meson.current_build_dir() / 'tests' / 'builddir',
-        meson.current_source_dir() / 'tests' / 'io.elementary.Files.PortalTester.yml',
-        '--force-clean'
-    ],
-    capture: true,
-    output: 'io.elementary.Files.PortalTester'
-)
+#custom_target(
+#    'portal-tester',
+#    command: [
+#        'flatpak-builder',
+#        '--user',
+#        '--install',
+#        meson.current_build_dir() / 'tests' / 'builddir',
+#        meson.current_source_dir() / 'tests' / 'io.elementary.Files.PortalTester.yml',
+#        '--force-clean'
+#    ],
+#    capture: true,
+#    output: 'io.elementary.Files.PortalTester'
+#)
