--- /data/20_elementary.pantheon.wm.gschema.override	2020-05-01 01:23:37.000000000 +0300
+++ /data/20_elementary.pantheon.wm.gschema.override	2020-05-02 13:09:22.368861666 +0300
@@ -1,6 +1,10 @@
 [org.gnome.mutter:Pantheon]
 dynamic-workspaces = true
 attach-modal-dialogs = false
-button-layout = 'close:maximize'
 edge-tiling = true
+
+[org.gnome.desktop.wm.preferences]
+button-layout = 'close:maximize'
+
+[org.gnome.desktop.interface]
 enable-animations = true
