--- /src/Indicator.vala	2018-12-12 04:04:37.000000000 +0300
+++ /src/Indicator.vala	2019-01-28 23:24:43.397840400 +0300
@@ -56,9 +56,31 @@
             main_grid = new Gtk.Grid ();
             main_grid.halign = Gtk.Align.CENTER;
             main_grid.valign = Gtk.Align.START;
-            main_grid.attach (calendar, 0, 0);
-            main_grid.attach (new Wingpanel.Widgets.Separator (), 0, 2);
-            main_grid.attach (settings_button, 0, 3);
+            main_grid.attach (calendar, 0, 1);
+            main_grid.attach (new Wingpanel.Widgets.Separator (), 0, 3);
+            main_grid.attach (settings_button, 0, 4);
+
+            GLib.Settings settings = new GLib.Settings ("io.elementary.desktop.wingpanel.datetime");
+            if (settings.get_boolean ("enable-time-break")) {
+                var break_grid = new Widgets.TimeBreak ();
+                main_grid.attach (break_grid, 0, 0);
+            }
+
+            settings.changed["enable-time-break"].connect(() => {
+                warning ("changed time breake widget");
+                if (settings.get_boolean("enable-time-break")) {
+                    warning ("changed time breake widget 2");
+                    var break_grid = new Widgets.TimeBreak ();
+                    break_grid.show_all ();
+                    main_grid.attach (break_grid, 0, 0);
+                } else {
+                    warning ("changed time breake widget 3");
+                    var exist_widget = main_grid.get_child_at (0,0);
+                    if (exist_widget != null) {
+                        exist_widget.destroy ();
+                    }
+                }
+            });
 
             calendar.day_double_click.connect (() => {
                 close ();
@@ -110,7 +132,7 @@
 
         event_grid = new Gtk.Grid ();
         event_grid.orientation = Gtk.Orientation.VERTICAL;
-        main_grid.attach (event_grid, 0, 1);
+        main_grid.attach (event_grid, 0, 2);
 
         foreach (var e in events) {
             var menuitem_icon = new Gtk.Image.from_icon_name (e.get_icon (), Gtk.IconSize.MENU);
--- /data/io.elementary.desktop.wingpanel.datetime.gschema.xml	2018-12-12 04:04:37.000000000 +0300
+++ /data/io.elementary.desktop.wingpanel.datetime.gschema.xml	2019-01-27 02:26:52.678320000 +0300
@@ -11,6 +11,21 @@
             <summary>Show week numbers.</summary>
             <description>If this is set to true, week numbers will be shown in the calendar.</description>
         </key>
+        <key type="b" name="enable-time-break">
+            <default>true</default>
+            <summary>Enable time break.</summary>
+            <description>Enable time break.</description>
+        </key>
+        <key type="i" name="break-time-value">
+            <default>15</default>
+            <summary>Break time value.</summary>
+            <description>Break time value.</description>
+        </key>
+        <key type="i" name="work-time-value">
+            <default>45</default>
+            <summary>Work time value.</summary>
+            <description>Work time value.</description>
+        </key>
         <key name="clock-format" enum="io.elementary.desktop.wingpanel.datetime.clock-formats">
             <default>"12h"</default>
             <summary>Whether the clock displays in 12h or 24h format</summary>
--- /meson.build	2018-12-12 04:04:37.000000000 +0300
+++ /meson.build	2019-01-28 16:23:41.936141700 +0300
@@ -18,6 +18,7 @@
     'src/DateRange.vala',
     'src/Indicator.vala',
     'src/Widgets/PanelLabel.vala',
+    'src/Widgets/TimeBreak.vala',
     'src/Widgets/calendar/Calendar.vala',
     'src/Widgets/calendar/CalendarModel.vala',
     'src/Widgets/calendar/CalendarView.vala',
