--- /src/MainView.vala
+++ /src/MainView.vala
@@ -94,6 +94,13 @@ public class DateTime.MainView : Gtk.Gri
         week_number_switch.valign = Gtk.Align.CENTER;
         week_number_switch.halign = Gtk.Align.START;
 
+        var date_label = new Gtk.Label (_("Show the date in the panel:"));
+        date_label.xalign = 1;
+
+        var date_switch = new Gtk.Switch ();
+        date_switch.valign = Gtk.Align.CENTER;
+        date_switch.halign = Gtk.Align.START;
+
         column_spacing = 12;
         row_spacing = 12;
         attach (time_format_label, 0, 0);
@@ -104,6 +111,8 @@ public class DateTime.MainView : Gtk.Gri
         attach (network_time_switch, 1, 2);
         attach (week_number_label, 0, 3);
         attach (week_number_switch, 1, 3);
+        attach (date_label, 0, 4);
+        attach (date_switch, 1, 4);
         attach (time_picker, 2, 2);
         attach (date_picker, 3, 2);
         show_all ();
@@ -116,9 +125,12 @@ public class DateTime.MainView : Gtk.Gri
         if (schema == null) {
             week_number_label.visible = false;
             week_number_switch.visible = false;
+            date_label.visible = false;
+            date_switch.visible = false;
         } else {
             wingpanel_settings = new GLib.Settings ("io.elementary.desktop.wingpanel.datetime");
             wingpanel_settings.bind ("show-weeks", week_number_switch, "active", SettingsBindFlags.DEFAULT);
+            wingpanel_settings.bind ("clock-show-date", date_switch, "active", SettingsBindFlags.DEFAULT);
         }
 
         bool syncing_datetime = false;
