--- /src/Widgets/Panel.vala	2019-03-18 19:38:23.000000000 +0300
+++ /src/Widgets/Panel.vala	2019-04-03 00:23:52.809295000 +0300
@@ -83,7 +83,7 @@
         }
 
         uint32 time = event.time;
-        
+
 #if HAS_GTK320
         window.get_display ().get_default_seat ().ungrab ();
 #else
@@ -125,6 +125,8 @@
         IndicatorEntry? sibling = null;
 
         switch (current.base_indicator.code_name) {
+            case "places-indicator":
+            case "workspaces-indicator":
             case Indicator.APP_LAUNCHER:
                 var children = left_menubar.get_children ();
                 int index = children.index (current);
@@ -140,6 +142,7 @@
                 }
 
                 break;
+            case "meteo-indicator":
             case Indicator.DATETIME:
                 var children = center_menubar.get_children ();
                 int index = children.index (current);
@@ -179,6 +182,8 @@
         IndicatorEntry? sibling = null;
 
         switch (current.base_indicator.code_name) {
+            case "places-indicator":
+            case "workspaces-indicator":
             case Indicator.APP_LAUNCHER:
                 var children = left_menubar.get_children ();
                 int index = children.index (current);
@@ -194,6 +199,7 @@
                 }
 
                 break;
+            case "meteo-indicator":
             case Indicator.DATETIME:
                 var children = center_menubar.get_children ();
                 int index = children.index (current);
@@ -233,10 +239,13 @@
         var indicator_entry = new IndicatorEntry (indicator, popover_manager);
 
         switch (indicator.code_name) {
+            case "places-indicator":
+            case "workspaces-indicator":
             case Indicator.APP_LAUNCHER:
                 indicator_entry.set_transition_type (Gtk.RevealerTransitionType.SLIDE_RIGHT);
                 left_menubar.add (indicator_entry);
                 break;
+            case "meteo-indicator":
             case Indicator.DATETIME:
                 indicator_entry.set_transition_type (Gtk.RevealerTransitionType.SLIDE_DOWN);
                 center_menubar.add (indicator_entry);
