--- /src/Widgets/CategoryFlowBox.vala	2020-04-30 23:47:16.000000000 +0300
+++ /src/Widgets/CategoryFlowBox.vala	2020-05-02 14:00:20.255434413 +0300
@@ -38,11 +38,11 @@
 
             flowbox = new Gtk.FlowBox ();
             flowbox.activate_on_single_click = true;
-            flowbox.column_spacing = 12;
-            flowbox.row_spacing = 12;
+            flowbox.column_spacing = 6;
+            flowbox.row_spacing = 6;
             flowbox.homogeneous = true;
             flowbox.min_children_per_line = 5;
-            flowbox.max_children_per_line = 5;
+            flowbox.max_children_per_line = 8;
             flowbox.selection_mode = Gtk.SelectionMode.NONE;
             flowbox.vexpand = true;
 
--- /src/Widgets/CategoryIcon.vala	2020-04-30 23:47:16.000000000 +0300
+++ /src/Widgets/CategoryIcon.vala	2020-05-02 14:00:55.468183983 +0300
@@ -37,7 +37,6 @@
 
         var layout = new Gtk.Grid ();
         layout.halign = Gtk.Align.CENTER;
-        layout.margin = 6;
         layout.orientation = Gtk.Orientation.VERTICAL;
         layout.add (icon);
         layout.add (plug_name);
