--- /src/meson.build	2023-04-04 20:39:57.000000000 +0300
+++ /src/meson.build	2023-08-28 19:42:05.261944407 +0300
@@ -39,32 +39,34 @@
 )
 
 # tests
+xvfb_run = find_program('xvfb-run', required: false)
+if xvfb_run.found()
+    test_env = [
+        'G_TEST_SRCDIR=' + meson.current_source_dir(),
+        'G_TEST_BUILDDIR=' + meson.current_build_dir(),
+        'GSETTINGS_SCHEMA_DIR=' + meson.project_build_root() / 'data',
+        'GSETTINGS_BACKEND=memory',
+        'GIO_USE_VFS=local'
+    ]
 
-test_env = [
-    'G_TEST_SRCDIR=' + meson.current_source_dir(),
-    'G_TEST_BUILDDIR=' + meson.current_build_dir(),
-    'GSETTINGS_SCHEMA_DIR=' + meson.project_build_root() / 'data',
-    'GSETTINGS_BACKEND=memory',
-    'GIO_USE_VFS=local'
-]
+    add_test_setup(
+    'headless',
+    exe_wrapper: [ xvfb_run, '-a', '-s', '-screen 0 1024x768x24 -noreset' ],
+    is_default: true
+    )
 
-add_test_setup(
-  'headless',
-  exe_wrapper: [ find_program('xvfb-run'), '-a', '-s', '-screen 0 1024x768x24 -noreset' ],
-  is_default: true
-)
-
-app_test = executable(
-    meson.project_name() + '.tests.application',
-    terminal_sources + 'tests/Application.vala',
-    dependencies: terminal_deps,
-    vala_args: [ '--define', 'TESTS']
-)
+    app_test = executable(
+        meson.project_name() + '.tests.application',
+        terminal_sources + 'tests/Application.vala',
+        dependencies: terminal_deps,
+        vala_args: [ '--define', 'TESTS']
+    )
 
-test(
-    'Application',
-    app_test,
-    env: test_env,
-    protocol: 'tap',
-    depends: test_schemas
-)
+    test(
+        'Application',
+        app_test,
+        env: test_env,
+        protocol: 'tap',
+        depends: test_schemas
+    )
+endif
