--- /src/Application.vala	2023-08-01 23:46:55.000000000 +0300
+++ /src/Application.vala	2023-08-21 22:29:33.458369806 +0300
@@ -66,8 +66,6 @@
             service_settings = new GLib.Settings (Constants.PROJECT_NAME + ".services");
             privacy_settings = new GLib.Settings ("org.gnome.desktop.privacy");
 
-            Environment.set_variable ("GTK_USE_PORTAL", "1", true);
-
             GLib.Intl.setlocale (LocaleCategory.ALL, "");
             GLib.Intl.bindtextdomain (Constants.GETTEXT_PACKAGE, Constants.LOCALEDIR);
             GLib.Intl.bind_textdomain_codeset (Constants.GETTEXT_PACKAGE, "UTF-8");
