--- /meson.build	2018-12-19 15:00:05.000000000 +0300
+++ /meson.build	2019-03-03 01:45:09.755219300 +0300
@@ -23,6 +23,12 @@
     c_name: 'as'
 )
 
+vala_args = []
+
+if get_option('bluetooth')
+    vala_args += ['--define=ENABLE_BLUETOOTH']
+endif
+
 shared_module(
     meson.project_name(),
     'src/Indicator.vala',
@@ -52,6 +58,7 @@
         wingpanel_dep
     ],
     install: true,
+    vala_args: vala_args,
     install_dir : wingpanel_dep.get_pkgconfig_variable('indicatorsdir')
 )
 
--- /src/Widgets/MprisWidget.vala	2018-12-19 15:00:05.000000000 +0300
+++ /src/Widgets/MprisWidget.vala	2019-03-03 02:07:32.015587700 +0300
@@ -21,11 +21,13 @@
 
     AppInfo? default_music;
     ClientWidget default_widget;
-    ClientWidget bluetooth_widget;
     HashTable<string,ClientWidget> ifaces;
     public signal void close ();
 
+#if ENABLE_BLUETOOTH
+    ClientWidget bluetooth_widget;
     public Sound.Services.ObjectManager object_manager;
+#endif
 
     public MprisWidget() {
         Object (orientation: Gtk.Orientation.VERTICAL, spacing: 1);
@@ -49,6 +51,7 @@
             pack_start(default_widget, false, false, 0);
         }
 
+#if ENABLE_BLUETOOTH
         object_manager = new Services.ObjectManager ();
         object_manager.bind_property ("has-object", this, "visible", GLib.BindingFlags.SYNC_CREATE);
 
@@ -81,6 +84,7 @@
                 }
             }
         });
+#endif
 
         show_all();
     }
--- /src/Widgets/MprisGui.vala	2018-12-19 15:00:05.000000000 +0300
+++ /src/Widgets/MprisGui.vala	2019-03-03 01:39:49.845583600 +0300
@@ -124,6 +124,8 @@
         Object (client: mpris_client);
     }
 
+#if ENABLE_BLUETOOTH
+
     /**
      * Create a new ClientWidget for bluetooth controls
      *
@@ -139,6 +141,7 @@
 
         update_controls ();
     }
+#endif
 
     /**
      * Create a new ClientWidget for the default player
