--- /src/Shortcuts/Settings.vala	2020-04-02 02:34:57.000000000 +0300
+++ /src/Shortcuts/Settings.vala	2020-05-22 19:51:12.530570696 +0300
@@ -72,10 +72,7 @@
             if (!valid (schema, key))
                 return (Shortcut) null;
 
-            if (schema == Schema.MEDIA)
-                return new Shortcut.parse (schemas[schema].get_string (key));
-            else
-                return new Shortcut.parse ((schemas[schema].get_strv (key)) [0]);
+            return new Shortcut.parse ((schemas[schema].get_strv (key)) [0]);
         }
 
         public bool set_val (Schema schema, string key, Shortcut sc) {
@@ -83,11 +80,7 @@
                 return false;
             }
 
-            if (schema == Schema.MEDIA) {
-                schemas[schema].set_string (key, sc.to_gsettings ());
-            } else {
-                schemas[schema].set_strv (key, {sc.to_gsettings ()});
-            }
+            schemas[schema].set_strv (key, {sc.to_gsettings ()});
 
             return true;
         }
