--- /src/Shortcuts/List.vala	2022-10-22 01:09:45.000000000 +0300
+++ /src/Shortcuts/List.vala	2023-08-16 15:18:32.449958717 +0300
@@ -109,7 +109,6 @@
             add_action (ref launchers_group, Schema.MEDIA, _("Email"), "email");
             add_action (ref launchers_group, Schema.MEDIA, _("Home Folder"), "home");
             add_action (ref launchers_group, Schema.MEDIA, _("Music"), "media");
-            add_action (ref launchers_group, Schema.MEDIA, _("Terminal"), "terminal");
             add_action (ref launchers_group, Schema.MEDIA, _("Internet Browser"), "www");
 
             media_group = {};
