--- /lib/Widgets/Settings.vala	2020-07-01 19:27:01.000000000 +0300
+++ /lib/Widgets/Settings.vala	2021-02-27 15:12:05.338387913 +0300
@@ -42,7 +42,14 @@
                 if (_prefers_color_scheme == null) {
                     setup_prefers_color_scheme ();
                 }
-                return _prefers_color_scheme;
+
+                if (_prefers_color_scheme == ColorScheme.DARK || _prefers_color_scheme == ColorScheme.LIGHT) {
+                    return _prefers_color_scheme;
+                } else {
+                    var gtk_settings = Gtk.Settings.get_default ();
+
+                    return gtk_settings.gtk_application_prefer_dark_theme ? ColorScheme.DARK : ColorScheme.LIGHT;
+                }
             }
             private set {
                 _prefers_color_scheme = value;
