--- /meson.build	2021-09-23 04:31:05.000000000 +0300
+++ /meson.build	2021-09-30 13:16:48.652861899 +0300
@@ -68,7 +68,6 @@
 libxml_dep = dependency('libxml-2.0', version: '>=2.6.32')
 rest_dep = dependency('rest-0.7', version: '>=0.7')
 sqlite3_dep = dependency('sqlite3', version: '>=3.5.9')
-webkit2gtk_dep = dependency('webkit2gtk-4.0', version: '>=2.0.0')
 gstreamer_plugins_base_vapi = meson.get_compiler('vala').find_library('gstreamer-base-1.0')
 gstreamer_plugins_base_dep = declare_dependency(dependencies: [gstreamer_plugins_base_vapi])
 lconv_dep = meson.get_compiler('vala').find_library('LConv', dirs: join_paths(meson.current_source_dir(), 'vapi'))
@@ -100,7 +99,6 @@
     libxml_dep,
     rest_dep,
     sqlite3_dep,
-    webkit2gtk_dep,
     lconv_dep,
     linux_dep,
 ]
@@ -142,6 +140,7 @@
 subdir('src')
 subdir('thumbnailer')
 if get_option('plugins')
+    dependencies += dependency('webkit2gtk-4.0', version: '>=2.0.0')
     subdir('plugins')
 endif
 subdir('po')
